render/gui/src/main/java/net/flintmc/render/gui/windowing/Window.java

```diff
@@ -105,6 +106,14 @@
    */
   boolean isKeyPressed(Key key);
 
+  /**
+   * Retrieves a set of all currently pressed keys by the user. The returned set is mutable and will
+   * be modified whenever a key is either pressed or released.
+   *
+   * @return A non-null mutable set with all currently pressed keys
+   */
+  Set<Key> getPressedKeys();
```

Comment:
If the `Set` is mutable, what happens if the user manually inserts a key? I think an immutable set would the be correct choice - they still reflect changes in their original sets, but can't be modified by the user